### PR TITLE
Module addition for multiple AZ

### PR DESCRIPTION
**Pull request by mrzta**
Add multiple AZ's to make the redis cache service more resilient

### CODE REVIEW

File: aks/redis/variables.tf [hunk 1]
```diff
@@ -95,6 +95,11 @@ variable "azure_enable_monitoring" {
   default = true
 }
 
+variable "redis_replica_zones" {
```
Comment (thomasleese) [inline aks/redis/variables.tf:98]:
```suggestion
variable "azure_zones" {
```

I've been naming any Azure-specific variables with `azure_` to keep them consistent.

**Comment by mrzta** [comment]
[Spike: Redis resilience](https://trello.com/c/9PqRQLbB/397-spike-redis-resilience)

File: aks/redis/variables.tf [hunk 2]
```diff
@@ -95,6 +95,11 @@ variable "azure_enable_monitoring" {
   default = true
 }
 
+variable "azure_redis_replica_zones" {
```
Comment (thomasleese) [inline aks/redis/variables.tf:98]:
```suggestion
variable "azure_zones" {
```

We don't need `redis` as we're already within the Redis module.

I'd also suggest `azure_zones` without `replica` so it matches the name of the argument we pass in to the`azurerm_redis_cache` resource, just makes it clear that we're not doing anything special with the argument, but I don't really mind on that.

**Review by thomasleese — APPROVED** [review]

File: aks/redis/resources.tf [hunk 3]
```diff
@@ -22,6 +22,7 @@ resource "azurerm_redis_cache" "main" {
   minimum_tls_version           = var.azure_minimum_tls_version
   public_network_access_enabled = var.azure_public_network_access_enabled
   redis_version                 = var.server_version
+  zones                         = var.azure_zones
```
Comment (RMcVelia) [inline aks/redis/resources.tf:25]: I get, Inappropriate value for attribute "zones": set of string required.
Also, zones not valid for standard, so only if premium.

Comment (mrzta) [reply aks/redis/resources.tf]: resolved the data type issues and introduced a condition

File: aks/redis/variables.tf [hunk 4]
```diff
@@ -95,6 +95,12 @@ variable "azure_enable_monitoring" {
   default = true
 }
 
+variable "azure_zones" {
```
Comment (saliceti) [inline aks/redis/variables.tf:98]: Do we really need an input variable here? We could set it to all zones by default when the SKU allows it
ie when it's not Basic or Standard

Comment (mrzta) [reply aks/redis/variables.tf:98]: thats true could set the default to 1.2 only as per the dev deployment

Comment (mrzta) [reply aks/redis/variables.tf:98]: updated the default to 2 zones

Comment (mrzta) [reply aks/redis/variables.tf]: removed variable

**Review by RMcVelia — APPROVED** [review]

File: aks/redis/resources.tf [hunk 5]
```diff
@@ -22,6 +22,7 @@ resource "azurerm_redis_cache" "main" {
   minimum_tls_version           = var.azure_minimum_tls_version
   public_network_access_enabled = var.azure_public_network_access_enabled
   redis_version                 = var.server_version
+  zones                         = var.azure_sku_name != "Premium" ? null : ["1", "2"]
```
Comment (saliceti) [inline aks/redis/resources.tf:25]: As discussed, redis multi AZ is not only Premium, as per https://azure.microsoft.com/en-gb/pricing/details/cache/?cdn=disable
We suggested the logic should be "NOT Basic and NOT Standard"

Comment (mrzta) [reply aks/redis/resources.tf]: updated condition

**Pull request merged**